packages/block-editor/src/hooks/block-bindings.js

```diff
@@ -2,42 +2,164 @@
+
+const {
+	DropdownMenuV2: DropdownMenu,
+	DropdownMenuGroupV2: DropdownMenuGroup,
+	DropdownMenuCheckboxItemV2: DropdownMenuCheckboxItem,
+	DropdownMenuItemLabelV2: DropdownMenuItemLabel,
+	DropdownMenuItemHelpTextV2: DropdownMenuItemHelpText,
+	DropdownMenuSeparatorV2: DropdownMenuSeparator,
+} = unlock( componentsPrivateApis );
+
+const useToolsPanelDropdownMenuProps = () => {
+	const isMobile = useViewportMatch( 'medium', '<' );
+	return ! isMobile
+		? {
+				popoverProps: {
+					placement: 'left-start',
+					// For non-mobile, inner sidebar width (248px) - button width (24px) - border (1px) + padding (16px) + spacing (20px)
+					offset: 259,
+				},
+		  }
+		: {};
+};
+
+function BlockBindingsPanelDropdown( {
+	fieldsList,
+	addConnection,
+	removeConnection,
+	attribute,
+	binding,
+} ) {
+	const currentKey = binding?.args?.key;
+	return (
+		<>
+			{ Object.entries( fieldsList ).map( ( [ label, fields ] ) => (
+				<>
+					<DropdownMenuGroup key={ label }>
+						{ Object.keys( fieldsList ).length > 1 && (
+							<Text
+								className="block-editor-bindings__source-label"
+								upperCase
+								variant="muted"
+							>
+								{ label }
+							</Text>
+						) }
+						{ Object.entries( fields ).map( ( [ key, value ] ) => (
+							<DropdownMenuCheckboxItem
+								key={ key }
+								onClick={ () =>
+									addConnection( key, attribute )
+								}
+								name={ attribute + '-binding' }
+								value={ key }
+								checked={ key === currentKey }
+							>
+								<DropdownMenuItemLabel numberOfLines={ 1 }>
+									{ key }
+								</DropdownMenuItemLabel>
+								<DropdownMenuItemHelpText numberOfLines={ 1 }>
+									{ value }
+								</DropdownMenuItemHelpText>
+							</DropdownMenuCheckboxItem>
+						) ) }
+					</DropdownMenuGroup>
+					<DropdownMenuSeparator />
+				</>
+			) ) }
+			<ButtonGroup className="block-editor-bindings__clear-binding">
+				<Button
+					__next40pxDefaultSize
+					disabled={ ! binding }
+					accessibleWhenDisabled
+					onClick={ () => removeConnection( attribute ) }
+					variant="tertiary"
+				>
+					{ __( 'Clear' ) }
+				</Button>
+			</ButtonGroup>
+		</>
+	);
+}
+
+function BlockBindingsAttribute( {
+	fieldsList,
+	addConnection,
+	removeConnection,
+	attribute,
+	binding,
+} ) {
+	const { source: sourceName, args } = binding || {};
+	const sourceProps =
+		unlock( blocksPrivateApis ).getBlockBindingsSource( sourceName );
+	return (
+		<DropdownMenu
+			placement="left-start"
+			gutter={ 36 }
+			className="block-editor-bindings__popover"
+			trigger={
+				<Item>
+					<DropdownMenuItemLabel numberOfLines={ 1 }>
+						{ attribute }
+					</DropdownMenuItemLabel>
+					{ !! binding && (
+						<DropdownMenuItemHelpText
+							numberOfLines={ 1 }
+							className="block-editor-bindings__item-explanation"
+						>
+							{ args?.key || sourceProps?.label || sourceName }
+						</DropdownMenuItemHelpText>
```

Comment:
While this seems to work ok, please do not use `DropdownMenuItem*` subcomponents inside other components (like `Item`). DropdownMenu subcomponents are meant to be used together with other DropdownMenu subcomponents.

Reply:
Okay. I guess we need to use `Truncate` and `VStack` for this use case then.

Reply:
I changed it to use `VStack` and `Truncate` instead: [link](https://github.com/WordPress/gutenberg/pull/62880/files#diff-bff98e322c4269c50aa13e4b251beb9d53d2d0c3d4066406f735d2d49bed83a1R104-R116). Hope that's okay.

Reply:
It should be fine, although I haven't tested those changes on my machine